src/index.ts

```diff
@@ -1,60 +1,100 @@
-    let pushMsg = `Pushing ${imageToPush} to ${registry}`;
-    if (username) {
-        pushMsg += ` as ${username}`;
-    }
-    core.info(pushMsg);
+    // get Podman cli
+    const podman = await io.which("podman", true);
 
-    //check if images exist in podman's local storage
-    const checkImages = await execute(podman, ['images', '--format', 'json']);
+    let imageToPush = `${imageInput}:${tag}`;
 
-    const parsedCheckImages = JSON.parse(checkImages.stdout);
+    // check if image exist in Podman local registry
+    const isPresentInPodman: boolean = await checkImageInPodman(
+        imageToPush,
+        podman,
+    );
+
+    // check if image exist in Docker local registry and if exist pull the image to Podman
+    const isPresentInDocker: boolean = await pullImageFromDocker(
+        imageToPush,
+        podman,
+    );
 
-    // this is to temporarily solve an issue with the case-sensitive of the property field name. i.e it is Names or names??
-    const nameKeyMixedCase = parsedCheckImages[0] && Object.keys(parsedCheckImages[0]).find(key => 'names' === key.toLowerCase());
-    const imagesFound = parsedCheckImages.
-                            filter((image: string) => image[nameKeyMixedCase] && image[nameKeyMixedCase].find((name: string) => name.includes(`${imageToPush}`))).
-                            map((image: string ) => image[nameKeyMixedCase]);
+    // boolean value to check if pushed image is from Docker local registry
+    let isPushingDockerImage = false;
 
-    if (imagesFound.length === 0) {
-        //check inside the docker daemon local storage
-        await execute(podman, [ 'pull', `docker-daemon:${imageToPush}` ]);
+    if (isPresentInPodman && isPresentInDocker) {
+        const warningMsg = "Image found in Podman as well as in Docker local registry.";
+
+        let isPodmanImageLatest = false;
+        try {
+            isPodmanImageLatest = await isPodmanLocalImageLatest(
+                imageToPush,
+                podman,
+            );
+        }
+        catch (err) {
+            core.setFailed(err);
+        }
+
+        if (!isPodmanImageLatest) {
+            core.warning(`${warningMsg} Using Docker local registry's image as that is built latest`);
+            imageToPush = `docker.io/library/${imageToPush}`;
+            isPushingDockerImage = true;
+        }
+        else {
+            core.warning(`${warningMsg} Using Podman local registry's image as that is built latest`);
+        }
+    }
+    else if (isPresentInDocker) {
+        imageToPush = `docker.io/library/${imageToPush}`;
+        isPushingDockerImage = true;
     }
 
-    // push image
-    const registryPath = `${registry.replace(/\/$/, '')}/${imageToPush}`;
+    let pushMsg = `Pushing ${imageToPush} to ${registry}`;
+    if (username) {
+        pushMsg += ` as ${username}`;
+    }
+    core.info(pushMsg);
+
+    const registryPath = `${registry.replace(/\/$/, "")}/${imageInput}:${tag}`;
```

Comment:
I know you didn't change this, but it took me a second to figure out what the regex is for here (I would have used `endsWith`. You could make it clearer by using an intermediate constant:
```js
const registryWithoutTrailingSlash = registry.replace(/\/$/, "")
```